### ISSUE TITLE
How to override remote methods for relations?

### ISSUE THREAD

**Issue by offlinehacker** [strongloop/loopback#487]

**Comment by loay** [comment]
Hello @offlinehacker 
Is this issue still relevant or it is been resolved? 

@fabien  Thanks a lot for the clarification.

**Comment by loay** [comment]
Closing due to inactivity. If you are still running into problems, feel free to leave a comment and I will reopen the issue.

**Comment by amaurybrisou** [comment]
I do need to override User delete method, instead of deleting I want to update deleted field. And I want to keep the same path :  DELETE /api/User

How can i achieve this in 2016 :) ?

**Comment by cadesalaberry** [comment]
@amaurybrisou Got the same question here !

**Comment by doorty** [comment]
+1 ..trying to override /api/user/{id}/items

**Comment by jmwohl** [comment]
+1 I need to override a hasAndBelongsToMany relation method, /api/entity/{id}/relation/rel/{fk}... overriding the __link__relation function does not work, and I don't quite understand the above suggestions. Is there any documentation about this?

**Comment by loay** [comment]
Re-opening issue. ..

**Comment by Narrator** [comment]
+1

**Comment by gdeckert** [comment]
+1

**Comment by kennethlynne** [comment]
Tried to register a remote method on the url of the relation method and it still calls the relation method.
Tried to register it after the "attached" event was fired. 
Tried to override the method on the model itself (__create__relation)

Nothing seems to work. Wtf :+1: :)

**Comment by IgorDavy** [comment]
+1

**Comment by dktan** [comment]
My work around:
1. Disable that remote method.
2. Create a new remote method of your own, name it as the one you want to override.

**Comment by windyinwind** [comment]
@dktan if you disable remote method /api/user/{id}/items, create your own. How did you define remote method to use the same url pattern? (I mean show "api/user/{id}/items" in explorer and get the "id" in code). I search how to implement in docs, but with no luck.

**Comment by deepakrkris** [comment]
@nghiaht @kennethlynne @windyinwind @dktan @loay @superkhau , I am able to override the inbuilt remote methods of the relation as following:
- [ ] - attach the customized method only after the application has booted or started.
- [ ] - add a model event handler for 'booted' (or 'started', if event is emitted in server.js, when app.listen() is called)
```
const app = require('../../');
module.exports = function(Customer) {
  app.on('started', function() {
    const origFindOrders = Customer.prototype['__findById__orders'];
    Customer.prototype['__findById__orders'] = function() {
      //custom logic here
      origFindOrders.apply(this, arguments);
    };
  });
};
```

**Comment by deepakrkris** [comment]
@superkhau @bajtos , it would be helpful to add documentation on overriding inbuilt remote methods. I will do it this sprint.

**Comment by superkhau** [comment]
Connect to #443

**Comment by superkhau** [comment]
Actually, closing in favour of #443

**Comment by sebastianfelipe** [comment]
Hi. I have some situation here. I'd like the existence of the "documented" parameter to bypass swagger without lose the remoting functionalities. Well, I realised that I can run a boot script, selecting which endpoint deploy on swagger and which not, but, the issue is with related models, I cannot apply the documented property. I think is related with this issue, I'd like to override the by default "documented" attribute set as shared method. Is there a way to do that with related models?

Thanks!
